src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -155,7 +212,13 @@ def _elements_from_raw_segments(
         for seg in segments:
             # NOTE: end_of_file is block-like rather than point-like.
             # This is to facilitate better evaluation of the ends of files.
-            if seg.is_type("whitespace", "newline", "indent"):
+            # NOTE: This also allows us to include literal placeholders for
+            # whitespace only strings.
+            if seg.is_type("whitespace", "newline", "indent") or (
+                seg.is_type("placeholder")
+                and cast(TemplateSegment, seg).block_type == "literal"
+                and cast(TemplateSegment, seg).source_str.isspace()
```

Comment:
I noticed a few cases of things like this in the PR. Random thought -- worth creating helper functions for these to keep the conditions a bit shorter and more readable? Judgement call, because there aren't a LOT of these.

Reply:
Good question - I'll see what I can do. I hear you that I do this same check a few times.